### PR TITLE
Combobox structure

### PR DESCRIPTION
**Pull request by joneff**
Note: I moved button structure to `html/meta/button` and changed `ref` to `$ref` (which i might change again if it's an issue).

@rkirilov @tsvetomir

### CODE REVIEW

File: packages/html/meta/button.json [hunk 1]
```diff
@@ -198,14 +198,14 @@
         ],
         "children": [
             {
-                "name": "icon",
-                "ref": "icon",
+                "name": "button-icon",
+                "$ref": "icon",
```
Comment (tsvetomir) [inline packages/html/meta/button.json:202]: Nice tweak, makes it really easy to spot.

**Review by tsvetomir — APPROVED** [review]

**Comment by rkirilov** [comment]
@joneff we will discuss the structure with the team and will get back to you next week

**Comment by joneff** [comment]
@rkirilov @tsvetomir @kvelikov @dtopuzov  I've updated the json files. See if you like them now better than you did before.

**Comment by rkirilov** [comment]
> @rkirilov @tsvetomir @kvelikov @dtopuzov I've updated the json files. See if you like them now better than you did before.

Looks good to me, but let's wait for @kvelikov review

**Review by Juveniel — APPROVED** [review]

**Comment by joneff** [comment]
@kvelikov any thoughts?

**Comment by kvelikov** [comment]
> @kvelikov any thoughts?

I think the proposed structure covers all cases so we're good.

**Comment by dtopuzov** [comment]
Some comments on the testing side.

We plan to have something like `verifyRendering()` function in https://github.com/telerik/kendo-e2e/.
Here are some initial thoughts on the implementation:
- We need spec as npm package
- We will NOT refer to this package in kendo-e2e because we want to allow teams to be compatible with different versions of the spec at different times, we also do not want to release new kendo-e2e just because we have a bug fix in the spec.
- Each product team will refer to the spec [as we do it for a11y spec now](https://github.com/telerik/kendo/blob/master/package.json#L20)
- This means verifyRendering should take spec from the product repo and we need to pass the spec as param to `verifyRendering`
- Currently, we [pass a11y spec as json object ](https://github.com/telerik/kendo-e2e/blob/master/src/a11y/a11y-wct-spec.ts#L21)parameter to `getWCTSpecViolations` and we want something similar for rendering spec
- It will be great if we can bundle all rendering specs in single json file before we publish the npm pacakge (generate single json spec build time)

**Comment by joneff** [comment]
@dtopuzov, @tsvetomir -- for your viewing pleasure: `@progress/kendo-html-spec`.

There are zero dependencies and also a very thin exposure around json files.

**Pull request merged**